### PR TITLE
Enhance: add entity version number and modify update entity API to work with version

### PR DESCRIPTION
**Pull request by sadiqkhoja**
### Changes:

- Added version column in the entity_defs table
- OData Feed returns version number
- `forms/attachment/dataset.csv` also returns version number
- PATCH /entities/:uuid checks conflict if force flag is not provided

#### What has been done to verify that this works as intended?

Integration tests and manual testing with frontend

#### Why is this the best possible solution? Were any other approaches considered?

Other option was to add version in query parameter or body, but version in ETag and If-Match is a cleaner approach

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

None

#### Does this change require updates to the API documentation? If so, please update docs/api.md as part of this PR.

Will do it in a separate PR

#### Before submitting this PR, please make sure you have:

- [x] run `make test-full` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

**Comment by matthew-white** [comment]
It sounds like this PR closes #832. Does that sound right?

File: test/integration/api/entities.js [hunk 1]
```diff
@@ -829,13 +838,24 @@ describe('Entities API', () => {
...
-        .expect(501)
+        .expect(409)
         .then(({ body }) => {
-          body.code.should.equal(501.1);
+          body.code.should.equal(409.15);
+        });
+    }));
+
+    it('should reject if version does not match', testEntities(async (service) => {
+      // TODO: change logic around force flag and enforcing certain kinds of updates
```
Comment (ktuite) [inline test/integration/api/entities.js:852]: can remove this and above TODO comment now?

**Review by ktuite — APPROVED** [review]

**Comment by ktuite** [comment]
I was trying to use this branch and had some problem loading the dataset properties in frontend. Null source forms?? It may have nothing to do with this PR, but reverting it made the problem go away.

Screen Shot 2023-08-29 at 10 03 14 AM

File: test/integration/api/datasets.js [hunk 2]
```diff
@@ -1317,7 +1317,7 @@ describe('datasets and entities', () => {
               .then(({ headers, text }) => {
                 headers['content-disposition'].should.equal('attachment; filename="goodone.csv"; filename*=UTF-8\'\'goodone.csv');
                 headers['content-type'].should.equal('text/csv; charset=utf-8');
-                text.should.equal('name,label,first_name,age\n12345678-1234-4123-8234-123456789abc,Alice (88),Alice,88\n');
+                text.should.equal('name,label,version,first_name,age\n12345678-1234-4123-8234-123456789abc,Alice (88),1,Alice,88\n');
```
Comment (ktuite) [inline test/integration/api/datasets.js:1320]: One thing I noticed when trying to make a form that looked up the version, and based on what we're talking about in the team meeting on Aug 29, `name` and `label` have to be what they are, but should `version` be `__version`?

Comment (sadiqkhoja) [reply test/integration/api/datasets.js:1320]: I like `version` for the attachment csv

**Pull request merged**